policy/policy.go

```diff
@@ -30,47 +31,71 @@ type branchVals struct {
 	UpgradeFromVer string                // Versions to test package upgrades from
 	PCPrivate      bool                  // indicates whether package cloud repo is private
 	Branch         map[string]branchVals `copier:"-"`
+	ReviewCount    string
+	Convos         bool
+	Tests          []string
+	SourceBranch   string
 }
 
 // Policies models the config file structure. The config file may contain one or more repos.
 type Policies struct {
-	Description string
-	PCRepo      string
-	DHRepo      string
-	CSRepo      string
-	PackageName string
-	Reviewers   []string
-	ExposePorts string
-	Binary      string
-	Protected   []string
-	Goversion   string
-	Default     string              // The default git branch(master/main/anything else)
-	Repos       map[string]Policies // map of reponames to branchPolicies
-	Ports       map[string][]string
-	Branches    branchVals
+	Description     string
+	PCRepo          string
+	DHRepo          string
+	CSRepo          string
+	PackageName     string
+	Reviewers       []string
+	ExposePorts     string
+	Binary          string
+	Protected       []string
+	Goversion       string
+	Default         string              // The default git branch(master/main/anything else)
+	Repos           map[string]Policies // map of reponames to branchPolicies
+	RepoPolicy      map[string]RepoPolicy
```

Comment:
@alephnull do you think is appropriated to add this new attribute RepoPolicy on the Policies struct? or should we add it instead on the RepoPolicy. From my point of view in terms of logic for the struct it should go into policies since the RepoPolicy struct represents a SINGLE repo. 
BUT placing this attribute here will require that when using **gpac** bundle we feed the templates from Policies struct instead of RepoPolicy which will imply we need to pass policies  struct as argument to the render templates function.

`func (r *RepoPolicy) renderTemplate(bundleDir, path string, isDir bool)`

Reply:
No, this is a leaky abstraction. Policies models the config file and RepoPolicy models the needs of the template renderer. The goal is the decouple the config file format from the template renderer so that changes in the config file format can be managed such that it does not impact existing template rendering code.

Reply:
I understand, you explained it to me last pairing but I still had my doubts for this particular case, thanks for the clarification again. I will move this attribute to the RepoPolicy struct then and move forward with implementation